### PR TITLE
App 3 0 layout sidebar refactor

### PR DESCRIPTION
**Pull request by Kadee80**
This PR:

- Has some substantial refactoring to CSS modules and component layouts
- Introduces SideBar and child component NavPanel to render nav and subnav
- Introduces `currentNavPanelTask: 'connect'` | `currentNavPanelTask: 'upload' ` to interface state
 - ^ requires reducer/handler to set current nav panel, also open to the idea of making SideBar a class based component
- Connect Button, Run Button, Run Controls, Run Log work the same as `app-3-0` branch, this is slightly behind the demo branch from this week, Upload File and Disconnect buttons are in place to be connected to RPC/State
 - Dinosaur.py can be uploaded by running `store.dispatch({type: 'robot:LOAD_PROTOCOL', meta: {robotCommand:true}})` in console (to test run screen functionality)

*note* this PR may simply function as a code review before more comprehensive PR next week, just wanted some eyes on this before anyone does any interface facing work next week
*TODO* add in missing proptypes

### CODE REVIEW

**Comment by Kadee80** [comment]
Based on convo re: #324 will rebase once merged and check for linting errors

File: app/ui/interface/index.js [hunk 1]
```diff
@@ -8,12 +8,19 @@ const makeInterfaceActionName = (action) => makeActionName(NAME, action)
 const getModuleState = (state) => state[NAME]
 
 const INITIAL_STATE = {
-  isNavPanelOpen: false
+  isNavPanelOpen: false,
+  currentNavPanelTask: 'connect'
```
[the remark this answers is not in the document]
Comment (Kadee80) [reply app/ui/interface/index.js]: added to initial state. do we need it in test as well?

[the remark this answers is not in the document]
Comment (Kadee80) [reply app/ui/interface/index.js]: Also, given we fixed the test I will have at a reducer in the morning, just need to do some research first

File: app/ui/components/NavPanel.css [hunk 2]
```diff
@@ -0,0 +1,96 @@
...
+  text-align: left;
+  background-size: 1rem;
+  background-repeat: no-repeat;
+  background-position: 0.5rem center;
+  margin-bottom: 0.5rem;
+}
+
+.btn_upload {
+  composes: btn;
+  background-image: url(../../ui/img/icon_fileupload.svg);
```
Comment (mcous) [inline app/ui/components/NavPanel.css]: Would `url(../img/icon_fileupload.svg)` work here? I don't think we need to go all the way up to `app`

File: app/ui/components/NavPanel.css [hunk 3]
```diff
@@ -0,0 +1,96 @@
...
+}
+
+.btn_new {
+  composes: btn;
+  background-image: url(../../ui/img/icon_filenew.svg);
```
Comment (mcous) [inline app/ui/components/NavPanel.css]: Same comment as above (unnecessary upwards directory traversal)

File: app/ui/components/NavPanel.css [hunk 4]
```diff
@@ -0,0 +1,96 @@
...
+  font-size: 1rem;
+  line-height: 3rem;
+}
+
+.btn_run {
+  composes: btn;
+  background-color: transparent;
+
+  /* TODO: extract assets from soon to be updated sketch files */
+  background-image: url(../../ui/img/icon_fileupload.svg);
```
Comment (mcous) [inline app/ui/components/NavPanel.css]: Same comment as above (unnecessary upwards directory traversal)

File: app/ui/components/NavPanel.css [hunk 5]
```diff
@@ -0,0 +1,96 @@
...
+  text-transform: uppercase;
+  font-size: 1rem;
+  line-height: 3rem;
+}
+
+.btn_run {
+  composes: btn;
+  background-color: transparent;
+
+  /* TODO: extract assets from soon to be updated sketch files */
```
Comment (mcous) [inline app/ui/components/NavPanel.css]: A cool thing to do might be to create a React icon library (because SVGs are awesome and we can do this)

**Pull request merged**